conda-forge.yml

```diff
@@ -1,13 +1,17 @@
+azure:
+  build_id: 48
+  user_or_org: TileDB-Inc
+  project_name: CI
 build_platform:
   osx_arm64: osx_64
 conda_forge_output_validation: true
 github:
   branch_name: main
   tooling_branch_name: main
 provider:
-  linux_aarch64: default
+  linux_aarch64: azure
   win: azure
-test_on_native_only: true
+test: native_and_emulated
```

Comment:
Fixes a conda-smithy warning:

````
WARNING:conda_smithy.configure_feedstock:conda-forge.yml: $.test_on_native_only = True -> 'Test On Native Only' is deprecated.
This was used for disabling testing for cross-compiling.

```warning
This has been deprecated in favor of the top-level `test` field.
It is now mapped to `test: native_and_emulated`.
```
````